### PR TITLE
Models should be loaded when required (Fixes #34)

### LINKED ISSUE #34: Models should be loaded when required
When calling any other method than `get` on the `ModelController`, without a prior call to `get`, it has no effect, which is unexpected, i.e. when starting the server and a clients calls `getAll`, an empty list is returned. We therefore should make sure to call `loadModel` during operations.

### PR DESCRIPTION
**Pull request by ndoschek**
- Load all models in workspace for general requests (get all
models/modeluris)
- Fix create/update model and add error response
- Remove duplicate slash in workspace URI

### CODE REVIEW

File: com.eclipsesource.modelserver.emf/src/main/java/com/eclipsesource/modelserver/emf/common/ModelRepository.java [hunk 1]
```diff
@@ -89,12 +93,22 @@ public void removeModel(String modeluri) {
 		return Optional.ofNullable(this.models.get(uri));
 	}
 
+	private void loadAllModelsInWorkspace() {
```
Comment (edgarmueller) [inline com.eclipsesource.modelserver.emf/src/main/java/com/eclipsesource/modelserver/emf/common/ModelRepository.java]: Rename to `ensureModelsAreLoaded` or similar (the `ensure` bit is important). Use same name for flag and method (`all` seems superfluous).

File: com.eclipsesource.modelserver.emf/src/main/java/com/eclipsesource/modelserver/emf/configuration/ServerConfiguration.java [hunk 2]
```diff
@@ -41,6 +48,21 @@ public void setWorkspaceRoot(String workspaceRoot) {
...
+	public Set<String> getWorkspaceEntries() {
+		Set<String> filePaths = new HashSet<>();
+		
+		if (!this.workspaceRoot.isEmpty()) {
+			try (Stream<Path> paths = Files.walk(Paths.get(this.workspaceRoot))) {
+				paths
+					.filter(Files::isRegularFile)
+					.forEach(file -> filePaths.add(file.toString()));
+			} catch (IOException e) {
+				e.printStackTrace();
```
Comment (edgarmueller) [inline com.eclipsesource.modelserver.emf/src/main/java/com/eclipsesource/modelserver/emf/configuration/ServerConfiguration.java]: Handle exception gracefully

File: com.eclipsesource.modelserver.emf/src/main/java/com/eclipsesource/modelserver/emf/common/ModelRepository.java [hunk 3]
```diff
@@ -89,12 +93,22 @@ public void removeModel(String modeluri) {
...
+
+	private boolean isModelWorkspaceLoaded() {
+		return modelWorkspaceLoaded;
+	}
+
 	private URI getWorkspaceUri(String modeluri) {
 		String baseURL = serverConfiguration.getWorkspaceRoot();
 		if (!modeluri.startsWith(baseURL)) {
-			modeluri = baseURL + "/" + modeluri.replaceAll(" ", "");
+			modeluri = baseURL + modeluri.replaceAll(" ", "");
```
Comment (edgarmueller) [inline com.eclipsesource.modelserver.emf/src/main/java/com/eclipsesource/modelserver/emf/common/ModelRepository.java:106]: How do we know that `baseURL` is ending with a slash here?

**Review by edgarmueller — COMMENTED** [review]
Looks fine to me, I left some comments.

File: com.eclipsesource.modelserver.emf/src/main/java/com/eclipsesource/modelserver/emf/common/ModelRepository.java (hunk 1 above)
Comment (ndoschek) [reply com.eclipsesource.modelserver.emf/src/main/java/com/eclipsesource/modelserver/emf/common/ModelRepository.java]: Agreed!

File: com.eclipsesource.modelserver.emf/src/main/java/com/eclipsesource/modelserver/emf/common/ModelRepository.java (hunk 3 above)
Comment (ndoschek) [reply com.eclipsesource.modelserver.emf/src/main/java/com/eclipsesource/modelserver/emf/common/ModelRepository.java:106]: uri.getPath() (in ServerConfiguration.java) adds a trailing slash to the path. Could not find an actual proof, but I made various observations...

Comment (edgarmueller) [reply com.eclipsesource.modelserver.emf/src/main/java/com/eclipsesource/modelserver/emf/common/ModelRepository.java:106]: Ok, let's track that as a separate issue then.

**Review by edgarmueller — APPROVED** [review]

File: com.eclipsesource.modelserver.emf/src/main/java/com/eclipsesource/modelserver/emf/common/ModelRepository.java (hunk 3 above)
Comment (tortmayr) [reply com.eclipsesource.modelserver.emf/src/main/java/com/eclipsesource/modelserver/emf/common/ModelRepository.java:106]: Eugen reported issues that are probably related to this.

ERROR ResourceManager  - Could not load resource with URI: /home/eugen/Git/coffee-editor/backend/examples/SuperBrewer3000SuperBrewer3000.coffee

Seems like the last / is missing

**Pull request merged**